store/structured/mongo/store.go

```diff
@@ -70,16 +83,30 @@ func (o *Store) Initialize(ctx context.Context) error {
 }
 
 func (o *Store) GetRepository(collection string) *Repository {
+	if o.client == nil {
+		// TODO: TK - should this return an error instead?
+		return nil
+	}
+
 	return NewRepository(o.GetCollection(collection))
 }
 
 func (o *Store) GetCollection(collection string) *mongoDriver.Collection {
+	if o.client == nil {
+		// TODO: TK - should this return an error instead?
+		return nil
+	}
+
 	db := o.client.Database(o.config.Database)
 	prefixed := fmt.Sprintf("%s%s", o.config.CollectionPrefix, collection)
 	return db.Collection(prefixed)
 }
 
 func (o *Store) Ping(ctx context.Context) error {
+	if o.client == nil {
```

Comment:
Perhaps `panic` is better, because this is not meant to be called before the store is initialized.